src/main.c

```diff
@@ -3027,6 +3028,16 @@ int main(int argc, char **argv)
         dispose_resolve_tag_arg(&resolve_tag_arg);
         yoml_free(yoml, NULL);
     }
+
+    {
+        const char *dir = dirname(h2o_socket_buffer_mmap_settings.fn_template);
+        struct stat st;
+        if (!(stat(dir, &st) == 0 && S_ISDIR(st.st_mode) && access(dir, W_OK) == 0)) {
+            fprintf(stderr, "temp-buffer-path: '%s' is not a writable directory\n", dir);
+            return EX_CONFIG;
+        }
+    }
```

Comment:
* How about actually calling `mkstemp` to make sure that a file can be created? I think that would make things simpler.
* I think we might want to move the logic (of checking if a temporary file can be created) into memory.c.

Reply:
It makes sense.

Also, I think using `dirname()` to format error messages is too smart so that it might hide what exactly occurred. So I've changed its error messages to show simply what h2o tried.

Sow how about this? a78a3c7d54e254e8f23cb75c780832356c106e6a